This patch adds the current XXH3 checksum algorithms to rsync. They
are not yet finalized, so do NOT use this in a distributed rsync yet!

To use this patch, run these commands for a successful build:

    patch -p1 <patches/xxh3.diff
    ./configure                         (optional if already run)
    make

based-on: 1bdf68b905b246ef4eb2959ba5e266005f761a8c
diff --git a/checksum.c b/checksum.c
--- a/checksum.c
+++ b/checksum.c
@@ -26,11 +26,13 @@
  * with this program; if not, visit the http://fsf.org website.
  */
 
+#define XXH_INLINE_ALL /* define to test pre-release XXH3 & XXH128 -- CURRENTLY UNSTABLE */
+
 #include "rsync.h"
 
 #ifdef SUPPORT_XXHASH
 #include "xxhash.h"
-# if XXH_VERSION_NUMBER >= 800
+# if XXH_VERSION_NUMBER >= 800 || defined XXH_INLINE_ALL
 #  define SUPPORT_XXH3 1
 # endif
 #endif
diff --git a/rsync.1.md b/rsync.1.md
--- a/rsync.1.md
+++ b/rsync.1.md
@@ -1474,6 +1474,8 @@ your home directory (remove the '=' for that).
     The checksum options that you may be able to use are:
 
     - `auto` (the default)
+    - `xxh128`
+    - `xxh3`
     - `xxh64` (aka xxhash)
     - `md5`
     - `md4`
